supervisor/resolution/checks/docker_config.py

```diff
@@ -0,0 +1,102 @@
+"""Helper to check if docker config for container needs an update."""
+
+from ...const import CoreState
+from ...coresys import CoreSys
+from ...docker.const import PropagationMode
+from ...docker.interface import DockerInterface
+from ..const import ContextType, IssueType, SuggestionType
+from ..data import Issue
+from .base import CheckBase
+
+
+def _check_container(container: DockerInterface) -> bool:
+    """Return true if container has a config issue."""
+    return any(
+        mount.get("Propagation") != PropagationMode.SLAVE.value
+        for mount in container.meta_mounts
+        if mount.get("Destination") == "/media"
+    )
+
+
+def setup(coresys: CoreSys) -> CheckBase:
+    """Check setup function."""
+    return CheckDockerConfig(coresys)
+
+
+class CheckDockerConfig(CheckBase):
+    """CheckDockerConfig class for check."""
+
+    async def run_check(self) -> None:
```

Comment:
Should this be executed the first time you add a media mount over the API as well?

Reply:
I mean tbh it will already have run in all likelihood. It's not limiting itself to when media mounts exist. Its just going to tell you that the default config has changed and the following things need a restart

Reply:
Fair point